pydm/display_module.py

```diff
@@ -1,40 +1,247 @@
+
+    Parameters
+    ----------
+    file : str
+        The path to a .ui file to load
+    macros : dict, optional
+        A dictionary of macro variables to supply to the
+        loaded display subclass.
+    args : list, optional
+        A list of command-line arguments to pass to the
+        loaded display subclass.
+    target : int
+        One of the ScreenTarget targets. PROCESS is only available when used
+        with PyDM Application for now.
+
+    Returns
+    -------
+    pydm.Display
+    """
+    if not is_pydm_app() and target == ScreenTarget.NEW_PROCESS:
+        logger.warning('New Process is only valid with PyDM Application. ' +
+                       'Falling back to ScreenTarget.DIALOG.')
+        target = ScreenTarget.DIALOG
+
+    if target == ScreenTarget.NEW_PROCESS:
+        # Invoke PyDM to open a new process here.
+        app = QApplication.instance()
+        app.new_pydm_process(file, macros=macros, command_line_args=args)
+        return None
+    else:
+        w = None
+        if file.endswith('.ui'):
+            w = load_ui_file(file, macros=macros)
+        else:
+            w = load_py_file(file, args=args, macros=macros)
+
+        if target == ScreenTarget.DIALOG:
+            w.show()
+
+        return w
+
+
+def load_ui_file(uifile, macros=None):
+    """
+    Load a .ui file, perform macro substitution, then return the resulting QWidget.
+
+    This is an internal method, users will usually want to use `open_file` instead.
+
+    Parameters
+    ----------
+    uifile : str
+        The path to a .ui file to load.
+    macros : dict, optional
+        A dictionary of macro variables to supply to the file
+        to be opened.
+
+    Returns
+    -------
+    QWidget
+    """
+    d = Display(macros=macros)
+    if macros is not None and len(macros) > 0:
+        f = macro.substitute_in_file(uifile, macros)
+    else:
+        f = uifile
+
+    klass, _ = uic.loadUiType(f)
+    # Add retranslateUi to Display class
+    d.retranslateUi = functools.partial(klass.retranslateUi, d)
+    d._loaded_file = uifile
+    klass.setupUi(d, d)
+    d.ui = d
+
+    return d
+
+
+def load_py_file(pyfile, args=None, macros=None):
+    """
+    Load a .py file, performs some sanity checks to try and determine
+    if the file actually contains a valid PyDM Display subclass, and if
+    the checks pass, create and return an instance.
+
+    This is an internal method, users will usually want to use `open_file` instead.
+
+    Parameters
+    ----------
+    pyfile : str
+        The path to a .ui file to load.
+    args : list, optional
+        A list of command-line arguments to pass to the
+        loaded display subclass.
+    macros : dict, optional
+        A dictionary of macro variables to supply to the
+        loaded display subclass.
+
+    Returns
+    -------
+    pydm.Display
+    """
+    # Add the intelligence module directory to the python path, so that
```

Comment:
Never noticed this before. What's an... intelligence module?

Reply:
It comes from here: https://github.com/slaclab/pydm/blob/7bfb9dcd2ebdaeaa23f46bc1af139df21c434261/pydm/utilities/display_loading.py#L58

An `intelligence module` in this case is the python code related to a certain display, which adds `intellicenge / logic` to the, most times, static ui.

Reply:
In the *very* early days of PyDM, before it even had a name, the idea of making displays that contained code was often discussed as "intelligent displays". Certain un-named managers even spoke of calling the project "IDM" for "Intelligent Display Manager"!  That all seemed way too pompous though, so we didn't go that way.